### ISSUE TITLE
Why mnist_autoencoder.prototxt contains two loss layers?

### ISSUE THREAD

**Issue by kingofoz** [BVLC/caffe#3104]
layer {
  name: "loss"
  type: "SigmoidCrossEntropyLoss"
  bottom: "decode1"
  bottom: "flatdata"
  top: "cross_entropy_loss"
  loss_weight: 1
}
layer {
  name: "decode1neuron"
  type: "Sigmoid"
  bottom: "decode1"
  top: "decode1neuron"
}
layer {
  name: "loss"
  type: "EuclideanLoss"
  bottom: "decode1neuron"
  bottom: "flatdata"
  top: "l2_error"
  loss_weight: 0
}

Please see above code. Thanks.

**Comment by kingofoz** [comment]
Any updates?

**Comment by seanbell** [comment]
From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:
>_Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.

**Comment by jeffdonahue** [comment]
Thanks @seanbell.  For the record it only has one loss layer in the sense that only the SigmoidCrossEntropyLoss layer has non-zero loss_weight.  The EuclideanLoss is just there for display purposes and may be removed with no effect on training (a la accuracy layers used in classification nets).

**Comment by kingofoz** [comment]
Thanks @jeffdonahue 
So you trained this model without using labels?

**Comment by kingofoz** [comment]
Hi @jeffdonahue
So you trained this model without using labels?

**Comment by jeffdonahue** [comment]
Yes, it's an autoencoder.  It's trained to reconstruct the data, not classify it.

**Comment by kingofoz** [comment]
Thanks very much!

**Comment by linrio** [comment]
And what the mean of ##cross_entropy_loss and ##l2_error ? which loss layer`s output can be regard as  the performance measurement ? cross_entropy_loss or l2_error ? 
my results are following:
`I0630 07:27:54.471556 22675 solver.cpp:237] Iteration 54800, loss = 4.28256e+07
I0630 07:27:54.471583 22675 solver.cpp:253]     Train net output #0: cross_entropy_loss = 4.28256e+07 (* 1 = 4.28256e+07 loss)
I0630 07:27:54.471595 22675 solver.cpp:253]     Train net output #1: l2_error = 8.69241e+06
I0630 07:27:54.471603 22675 sgd_solver.cpp:106] Iteration 54800, lr = 0.02
I0630 07:27:55.668200 22675 solver.cpp:237] Iteration 54900, loss = 2.05716e+07
I0630 07:27:55.668229 22675 solver.cpp:253]     Train net output #0: cross_entropy_loss = 2.05717e+07 (* 1 = 2.05717e+07 loss)
I0630 07:27:55.668241 22675 solver.cpp:253]     Train net output #1: l2_error = 6.32653e+06
I0630 07:27:55.668251 22675 sgd_solver.cpp:106] Iteration 54900, lr = 0.02
I0630 07:27:56.851256 22675 solver.cpp:459] Snapshotting to binary proto file mnist_autoencoder_iter_55000.caffemodel
I0630 07:27:56.959489 22675 sgd_solver.cpp:273] Snapshotting solver state to binary proto file mnist_autoencoder_iter_55000.solverstate
I0630 07:27:57.017945 22675 solver.cpp:321] Iteration 55000, loss = -5.28994e+06
I0630 07:27:57.017984 22675 solver.cpp:326] Optimization Done.
I0630 07:27:57.017992 22675 caffe.cpp:215] Optimization Done.`
